### PR TITLE
[Base Screens - Registration] VerifyCodeScreenBase

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-5108 .

#### Changes proposed in this Pull Request:

- Added VerifyCodeScreenBase component
- Added to the example project
- Added doc for VerifyCodeScreen

#### Screenshots / Screen Recording (if applicable)
Screenshot 2024-02-16 at 5 45 03 PM

verify-email

#### To Test:

- yarn start:example-android/ios

#### Any specific feedback you are looking for?

-

### CODE REVIEW

**Comment by JeffGreiner-eaton** [comment]
Question for anyone, should we open numeric keypad for the verify code field?

image

**Comment by JeffGreiner-eaton** [comment]
The handleOnNext and calling function codeValidator / verifyCode not called when next is pressed.

The send again (onResend) function call is working fine...
image

File: login-workflow/src/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    WorkflowCardActions,
+    WorkflowCardBody,
+    WorkflowCardHeader,
+    WorkflowCardInstructions,
+} from '../../components';
+import { HelperText, Text, TextInput } from 'react-native-paper';
+import { View } from 'react-native';
+import { useExtendedTheme } from '@brightlayer-ui/react-native-themes';
+
+/**
```
Comment (JeffGreiner-eaton) [inline login-workflow/src/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx:15]: you can follow whats been done for other components here...
@param {VerifyCodeScreenProps} props - props of VerifyCodeScreen......

File: login-workflow/docs/screens/verify-code.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+| ------------------------ | ------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------- |
+| codeValidator            | `(code: string) => boolean \| string` | A function to validate the input value                                                                                                                                  | required to be > 0 characters                   |
+| onResend                 | `() => void`                          | A function that is called when the resend link/button is clicked.                                                                                                       |                                                 |
+| resendInstructions       | `string`                              | Text to display ahead of the resend link/button.                                                                                                                        | `t('bluiAuth:VERIFY_CODE.RESEND_INSTRUCTIONS')` |
```
Comment (JeffGreiner-eaton) [inline login-workflow/docs/screens/verify-code.md:25]: you can remove the translation values and replace with the default values

**Review by surajeaton — COMMENTED** [review]
Please resolve the conflicts.

**Comment by manojleaton** [comment]
> The handleOnNext and calling function codeValidator / verifyCode not called when next is pressed.
> 
> The send again (onResend) function call is working fine... image

Have you tried passing callBack function to the prop `onNext` in `WorkflowCardActionProps`, it is working fine in my local. Please try adding this code in VerifyCodeScreenBaseExample 
WorkflowCardActionsProps={{
                showPrevious: true,
                showNext: true,
                previousLabel: 'Back',
                nextLabel: 'Next',
                canGoNext: true,
                canGoPrevious: true,
                onPrevious: (): void => navigation.navigate('Home'),
                onNext: (code: any) => console.log(code),
            }}

**Comment by ArshdeepSingh3-eaton** [comment]
Looks good

**Comment by JeffGreiner-eaton** [comment]
looks like only item left is the jsDoc section in verifyCodeScreenBase. I noticed that we have other screen base components checked in to the release/R31 branch that does not contain the same references as Ekta updated in the components directory. Are screen base components getting same jsDoc entries like here for example... https://github.com/etn-ccis/blui-react-native-workflows/blob/d3078dd767c97a3cd98e12c0cb53c530877404f3/login-workflow/src/components/SetPassword/SetPassword.tsx#L33

**Review by JeffGreiner-eaton — APPROVED** [review]

**Pull request merged**